app/mapping.py

```diff
@@ -70,16 +72,21 @@ def get_mapping(index_name, document_type):
         # than use index_name, we access the one and only value in the dictionary using next(iter).
         with logged_duration_for_external_request('es'):
             mapping_data = next(iter(es.indices.get_mapping(index=index_name, doc_type=document_type).values()))
-
-        # In ES <=5, there may be multiple mapping types per document type (kind-of) - this is confusing, but going
-        # away, see <https://www.elastic.co/guide/en/elasticsearch/reference/5.6/removal-of-types.html>. For us,
-        # document_type === mapping_type, and there will be only one per index.
-        return Mapping(mapping_data, mapping_type=document_type)
-
+    except NotFoundError as e:
+        if e.error == "type_missing_exception":
+            raise MappingNotFound("Document type '{}' is not valid in index '{}' - no mapping found.".format(
+                document_type, index_name))
+        else:
+            raise
     except StopIteration:
```

Comment:
Will line 74 ever throw `StopIteration` now? We will only ever want the first result from that `iter`. Relatedly, I'm getting a deprecation warning when I run the tests: `DeprecationWarning: generator 'logged_duration' raised StopIteration` so it might be nice to do something different there if we can...

Reply:
As for `StopIteration` handling I'd leave it in just for robustness.

Reply:
Looks like the StopIteration comes from the builtin `next` function, can't be doing much about that.

Reply:
L74 will raise StopIteration where the Elasticsearch server is version 5.4

Reply:
Running the tests against latest dmrunner `master` removed the deprecation warning.